musings/verification-machine.md

```diff
@@ -0,0 +1,243 @@
+
+# Verification Machinery
+
+## Verification From Ten Thousand Feet
+
+The picture below is yet another way to slice the [RATS architecture](https://www.rfc-editor.org/rfc/rfc9334.html#figure-1), which makes the (important) distinction between _"identity endorsements"_ (e.g., verification keys), and _"endorsed values"_ (e.g., certification status of the gadget, color, etc.), which is not explicit in RFC9334.
+
+This distinction becomes relevant when zooming in to the [appraisal process](#a-closer-look-to-the-appraisal-process) where one kind or the other is used depending on the verification "stage."
+
+```mermaid
+flowchart TD
+    R[(reference</br>values)]
+    Ev[(endorsed</br>values)]
+    Ei[(identity</br>endorsement)]
+    P[(appraisal policy</br>for evidence)]
+    VO([verifier</br>owner])
+    RFP([reference</br>value</br>provider])
+    E([endorser])
+    RP([relying</br>party])
+    A([attester])
+    e((Evidence))
+    a((Attestation</br>Result))
+    V([verifier])
+
+    subgraph " "
+    Ev
+    Ei
+    end
+
+    subgraph " "
+    R
+    end
+    
+    subgraph " "
+    P
+    end
+
+    VO --> P
+    RFP --> R
+    E --> Ev
+    E --> Ei
+    P --> V
+    R --> V
+    RP --> e
+    A --> e
+    Ev --> V
+    Ei --> V
+    V --> a
+    e --> V
+    A -.-> RP
+
+```
+
+## A Closer Look to the Appraisal Process
+
+The picture below provides a closer look in to the appraisal process, describing what a typical verification flow could look like.  The flow is broken down into four separate stages:
+
+* [Crypto verification](#crypto-verification)
+* [Reference values match](#reference-values-match)
+* [Endorsed values decoration](#endorsed-values-decoration)
+* [Attestation result assembly and signing](#attestation-result-assembly-and-signing)
+
+> **Notes**:
+>
+> * The appraisal policy is an input to all the processing stages
+> * All non-final processing stages can output a "failure" signal that short circuits the appraisal
+  
+```mermaid
+flowchart TD
```

Comment:
Nice picture, when it renders, I have a doubt.
Just minor clarifications/improvements, below!

`What is the scope of the Outer yellow boxes, is not clear to me, looking at the diagram??
In the centre if the scope is the Verifier entity, then we should label it to either Verifier or
Verifier actions then it all makes sense. Then in the abstraction Verifier Signing key can be folded in the big Verifier Box..

The color of the other boxes can then be kept other than yellow to indicate components external to the main verification logic.

Reply:
> `What is the scope of the Outer yellow boxes, is not clear to me, looking at the diagram??

It just highlights different logical elements.

> In the centre if the scope is the Verifier entity, then we should label it to either Verifier or Verifier actions then it all makes sense.

Sure, good point.

> Then in the abstraction Verifier Signing key can be folded in the big Verifier Box..

I think it's better to keep it outside the verifier box.